### PR TITLE
[Merge] Adjust ExecutionPayload.transaction type with new Transaction Union structure 

### PR DESCRIPTION
**Pull request by Nashatyrev**
## PR Description

This PR based on top of #4361

- create a dedicated `Transaction` Union structure for cleaner code 
- adjust `ExecutionPayload.transaction` with type `SszList<Transaction>`

## Fixed Issue(s)

Part of #4226 

Reduces number of failing ref tests from 633 to 170

## Documentation

- [x] I thought about documentation and added the `documentation` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: eth-reference-tests/src/referenceTest/java/tech/pegasys/teku/reference/ManualReferenceTestRunner.java [hunk 1]
```diff
@@ -30,7 +29,7 @@
  * <p>The test case is disabled as the tests run via the generated classes in CI, but it still runs
  * without removing the @Disabled in IntelliJ.
  */
-@Disabled
+// @Disabled
```
Comment (ajsutton) [inline eth-reference-tests/src/referenceTest/java/tech/pegasys/teku/reference/ManualReferenceTestRunner.java]: We should disable this again before merge so it doesn't run in CI.

**Review by ajsutton — APPROVED** [review]
LGTM. See comments on the SszUnion stuff from the other PR.

**Comment by ajsutton** [comment]
Confirmed this gets all the ssz_static reference tests passing which is great.

File: eth-reference-tests/src/referenceTest/java/tech/pegasys/teku/reference/ManualReferenceTestRunner.java (hunk 1 above)
Comment (Nashatyrev) [reply eth-reference-tests/src/referenceTest/java/tech/pegasys/teku/reference/ManualReferenceTestRunner.java]: Ups, thanks 👍

**Pull request merged**